### ISSUE TITLE
Comments are parsed (and acts weird)

### ISSUE THREAD

**Issue by Xendiadyon** [scssphp/scssphp#38]
We lately switched to scssphp/scssphp 1.0.3 (from leafo/scssphp) and found two issues, which we were able to trace back to one single problem by tedious debugging.

### Short Bug report
Please DO NOT parse anything inside of /* */ comments!

### Issue 1: Variables in Comments
We had a documentation inside a SCSS file on how to work with a mixin. Here, we had used some comment, which should be absolutely fine.
```CSS
/*
Usage:
@include mediaSwitch(background,red,green,blue);
or using lists:
$imgArea: 550px, 280px, 174px;
@include mediaSwitch(height,$imgArea...);
*/
```

When switching from leafo/scssphp to scssphp/scssphp, this comment was parsed and an error "can not find undeclared variable $imgArea" was thrown.
No, this is absolutely NOT desired!

### Issue 2: Versions in comments
This was great fun to find!
We implemented the tingle library. At the beginning, there was a comment, which should be absolutely fine:
```CSS
/* ----------------------------------------------------------- */
/* == tingle v0.15.1 */
/* ----------------------------------------------------------- */
```
*BUT:* In this comment, there was the version number `v0.15.1` which was parsed by scssphp/scssphp probably to something like: `v0 0.15 0.1`.

At the first sight, this should be no problem, as it is just inside of a comment, right?

However, we noted that *in certain cases*, a self-written Media-Query-Mixin `smallScreen` in a file which was imported **before** the tingle-SCSS-Subfile did not work:
```CSS
@include smallScreen {
  .navBar {
    display: none;
    border: 2px solid yellow; // works
  }
}
@include smallScreen {
  .navBar {
    display: block;
    border: 2px solid orange; // works
  }
}
.navBar
{
  @include smallScreen {
    border:2px solid green; // works
  }
  display:block;
  border:2px solid blue;
  @include smallScreen {
    border:2px solid blueviolet; // works
  }
  @include smallScreen {
    div
    {
      border:2px solid purple; // DOES NOT WORK
    }
  }
}
```

I can't tell why exactly this number-reformating led to such a strange behavior.
Alas, it doesn't matter.
If there is nothing parsed inside the comment section /* */, these two issues should not occur again.

Thank you :)

**Comment by robocoder** [comment]
Try using `//`.

Interpolation in multiline comments was implemented for greater compatibility with the Ruby scss implementation.

Reference: https://sass-lang.com/documentation/syntax/comments

**Comment by stof** [comment]
@robocoder but the comment reported in this issue is not using the interpolation syntax. The official Sass only supports interpolation there.

Also, you should stop talking about "Ruby scss" when referring to https://sass-lang.com/. The reference implementation is implemented in Dart now, not in Ruby. Ruby-sass has been deprecated in April 2018 (see http://sass.logdown.com/posts/7081811) and is EOL since March 2019 (see http://sass.logdown.com/posts/7828841-ruby-sass-is-unsupported).

**Comment by robocoder** [comment]
@stof I know dart sass is now the official implementation. However, scssphp used ruby scss as its reference implementation, so it's going to deviate more from the spec, e.g., support for `/*! ... */`.

**Comment by stof** [comment]
Well, the page you linked as justification for the behavior you implemented is explicitly saying that multiline comments support interpolation, not scss syntax.

**Comment by robocoder** [comment]
No, I included the link so we fix the current behaviour to match the spec.

**Comment by robocoder** [comment]
(Sorry if that wasn't clear. Otherwise I would have closed the issue.)

**Comment by Cerdic** [comment]
I can agree there is a bug somewhere here, cause I encountered it once while digging in another bug — and my bad didn't keep the faulty test case.

But at the moment I'm not able to reproduce with your comment whatever I'm trying to build a test case around it:

```
/*
Usage:
@include mediaSwitch(background,red,green,blue);
or using lists:
$imgArea: 550px, 280px, 174px;
@include mediaSwitch(height,$imgArea...);
*/

$imgArea: 3px;
.foo {
	color:red;
	width: $imgArea;
}
```

works perfectly giving

```
/*
 Usage:
 @include mediaSwitch(background,red,green,blue);
 or using lists:
 $imgArea: 550px, 280px, 174px;
 @include mediaSwitch(height,$imgArea...);
 */
.foo {
  color: red;
  width: 3px; }
```

Moreover, this is a more complex bug than exposed here as the parser is supposed to parse only interpolations: see https://github.com/scssphp/scssphp/blob/master/src/Parser.php#L1069

I can only suppose there is some unobvious borderline testcase here, but if you were able to provide us a simple test case this will really helpfull

**Comment by Cerdic** [comment]
@Xendiadyon I made some improvement in the comment parsing: avoiding any error on interpolation on an unknown var and any poisoning of the env when evaluating the interpolation in a comment.

Can you check if it's solving your issue, or otherwise work on a reproductible simple test case?

**Comment by Xendiadyon** [comment]
Using Version 1.0.4, it still throws an error.

**no error:** `// #{nth($imgArea,3)};`
**error thrown:** `/* #{nth($imgArea,3)}; */`

**Comment by stof** [comment]
@Xendiadyon the SCSS spec mandates that interpolation is supported in CSS comments (`//` is a SCSS comment)

**Comment by Cerdic** [comment]
Are you using a composer install or a git clone ?
The patch is not included in the 1.0.4
But on the master branch this is solved and the simple use case
```
/* #{nth($imgArea,3)}; */
```

is kept as is not producing any error

**Comment by Xendiadyon** [comment]
scssphp is a part of a greater project (Contao 4.4 LTS). The dependencies are resolved using the composer, yes.

**Comment by robocoder** [comment]
@Xendiadyon Can you confirm whether or not this is still a problem for you. This was included in the 1.0.5 release.

**Comment by Cerdic** [comment]
No answer is an agreement :)
